### PR TITLE
PaidCommand Bug

### LINKED ISSUE #402: BUG: Paying lessons that are equals causes lesson to disappear
## Steps to reproduce:
1. Create two lessons with the same date, time range, rates and cancelled, but different outstanding fees.

image

2. Pay one lesson such that both lessons have the same outstanding fee

Result: the lesson disappears

## Fix:
- Do not use the `Lesson#equals` method in `PaidCommand`
- create empty `TreeSet`. Then remove `toEdit` and add `editedLesson` 

Please follow the implementation of `LessonEditCommand`
- refactor to use `CommandUtil.getPerson`, `CommandUtil.getLesson` 
- instead of a local `createdEditedPerson` method, have a `createUpdatedLessons` method.

### PR DESCRIPTION
**Pull request by angkoonhwee**
Fixes #402

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/PaidCommand.java [hunk 1]
```diff
@@ -158,6 +142,24 @@ private static Lesson createEditedLesson(Lesson lessonToEdit, Money payment) thr
...
+     * Replaces lesson {@code toEdit} with lesson {@code edited} in {@code lessonList}.
+     *
+     * @param lessonList A list of lessons to update.
+     * @param paidLesson The paid lesson.
+     * @param lessonToPay The original lesson to be pay.
+     * @return A set of updated lessons with the lesson edited.
+     * @throws CommandException If the edited lesson results in a clash.
+     */
+    private Set<Lesson> createUpdatedLessons(List<Lesson> lessonList, Lesson paidLesson, Lesson lessonToPay)
+            throws CommandException {
```
Comment (lingshanng) [inline src/main/java/seedu/address/logic/commands/PaidCommand.java:155]:
```suggestion
    private Set<Lesson> createUpdatedLessons(List<Lesson> lessonList, Lesson paidLesson, Lesson lessonToPay)
             {
```

**Review by Xiaoyunnn — COMMENTED** [review]
still got bug eh... i paid for the first lesson but the 2nd lesson got updated
image

**Comment by eeliana** [comment]
> still got bug eh... i paid for the first lesson but the 2nd lesson got updated image

I think is because of the ordering; from our POV the lessons look identical but they are treated different backend @lingshanng

**Comment by lingshanng** [comment]
@Xiaoyunnn ya it's because of the ordering. You should be able to tell the difference between the lessons if you add homework.

**Review by lingshanng — APPROVED** [review]
LGTM

**Comment by lingshanng** [comment]
@Xiaoyunnn  ya it's the ordering, I don't think there's a way to fix the order. It's quite an obscure case so hopefully no one finds it an issue

**Pull request merged**